akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/Effect.scala

```diff
@@ -102,6 +105,27 @@ import akka.util.ccompat.JavaConverters._
    */
   def noReply(): ReplyEffect[Event, State] =
     none().thenNoReply()
+
+  /**
+   * Asynchronous command handling. The effect is run when the `CompletionStage` has been completed.
+   * Any incoming commands are stashed and processed later, after current command, when the `CompletionStage` has
+   * been completed.
+   *
+   * This can for example be used for retrieval of external information before validating the command.
+   */
+  def async(effect: CompletionStage[Effect[Event, State]]): Effect[Event, State] = {
```

Comment:
Do we need some extra warnings about closing over internal state here? In the Java side of things we expect that the state may be mutable, but otoh not mutated until this async command-handling completes so maybe not so dangerous after all. User could still end up messing around with `ActorContext` though?

Reply:
No other command is handled until the after this async effect so I don't think there is a problem with closing over the state. Messing with ActorContext from another thread should be documented by ActorContext itself, and we have currentActorThread checks.